### PR TITLE
Refactor getInitialData on OD Radio and Podcast pages to reduce repetition

### LINKED ISSUE #8747: Render Podcast Pages in Simorgh with current On demand Radio Page
**Describe the solution you'd like**
Add a new page type with regex for Podcast pages that renders using the on demand radio page.

Brand/Series route for Podcast page
- /{service}/podcasts/{brand_series_pid}
- /{service}/{var}/podcasts/{brand_series_pid}

Episode route for podcast page
- /{service}/podcasts/{brand_series_pid}/{episode_pid}
- /{service}/{var}/podcasts/{brand_series_pid}/{episode_pid}

- Route object needs setup for podcast page
- Path regex needs setup
- we need to setup the `getInitialData` function for podcast pages that uses the OD radio page (should we rename this page?)
- setup routes to fixture data for local development

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
Most of the ondemand work can be reused here:
- https://github.com/bbc/simorgh/tree/latest/src/app/routes/onDemandRadio
- https://github.com/bbc/simorgh/blob/latest/src/app/pages/OnDemandRadioPage/OnDemandRadioPage.jsx
-

### PR DESCRIPTION
**Pull request by pharingee**
**Overall change:**
Refactor getInitialData on OD Radio and Podcast pages to reduce repetition

**Code changes:**
- Create `radioPodcast/getInitialData.js` with shared `getInitialData` code.
- Use `radioPodcast/getInitialData.js` in `onDemandRadio` and `podcast` routes

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/radioPodcast/getInitialData/index.js [hunk 1]
```diff
@@ -66,7 +76,7 @@ export default async ({ path: pathname, pageType, service, toggles }) => {
     return {
       status,
       pageData: {
-        metadata: { type: 'On Demand Radio' },
+        metadata: { type: detailPageType },
```
Comment (rebeccamcginn) [inline src/app/routes/radioPodcast/getInitialData/index.js]: We need to pass through brandId for podcasts, required for recentEpisodes links

https://github.com/bbc/simorgh/pull/8766/files#diff-7a0fe662d3653d7e9cea8671ad812edc4182cc100b1e31fd03c20361b783c8ecR85

File: src/app/routes/radioPodcast/getInitialData/index.js [hunk 2]
```diff
@@ -15,19 +15,29 @@ import withRadioSchedule from '#app/routes/utils/withRadioSchedule';
...
+    const RADIO_PODCAST_MISSING_FIELD = isPodcast
+      ? PODCAST_MISSING_FIELD
+      : RADIO_MISSING_FIELD;
+    const DEFAULT_TOGGLE_VALUE = { enabled: false, value: isPodcast ? 8 : 4 };
+    const recentEpisodesKey = isPodcast
+      ? 'recentPodcastEpisodes'
+      : 'recentAudioEpisodes';
+    const getRecentEpisodesToggle = pathOr(DEFAULT_TOGGLE_VALUE, [
+      recentEpisodesKey,
+    ]);
```
Comment (ryanmccombe) [inline src/app/routes/radioPodcast/getInitialData/index.js]: Might be more readable to pass these values from the route file, eg:

```javascript
export default config => async ({ path: pathname, pageType, service, toggles }) => {
    const { detailPageType, missingFieldCode, etc } = config;
```

Or to keep the variants together here, eg:

```javascript
const config = pathname.includes('podcast') ? {
    detailPageType: 'Podcast',
    missingFieldCode: PODCAST_MISSING_FIELD,
    defaultToggleValue: { enabled: false, value: 8 },
    recentEpisodesKey: 'recentPodcastEpisodes`
} : {
    // Radio Values
}
```

**Review by ryanmccombe — APPROVED** [review]

File: src/app/routes/onDemandAudio/getInitialData/index.js [hunk 3]
```diff
@@ -15,19 +15,45 @@ import withRadioSchedule from '#app/routes/utils/withRadioSchedule';
...
 const getScheduleToggle = path(['onDemandRadioSchedule', 'enabled']);
-const getRecentEpisodesToggle = pathOr(DEFAULT_TOGGLE_VALUE, [
-  'recentAudioEpisodes',
-]);
+
+const getConfig = pathname => {
+  const detailPageType = pathname.includes('podcast')
+    ? 'Podcast'
+    : 'On Demand Radio';
+  const isPodcast = detailPageType === 'Podcast';
```
Comment (RichardPK) [inline src/app/routes/onDemandAudio/getInitialData/index.js:25]: Could `isPodcast` be assigned to `pathname.includes('podcast')` on ln22 and then use that for the `detailPageType` ternary, as it is in all the other ternaries?

File: src/app/routes/onDemandAudio/getInitialData/index.test.js [hunk 4]
```diff
@@ -16,7 +16,8 @@ describe('Get initial data for on demand radio', () => {
     jest.clearAllMocks();
   });
 
-  it('should return essential data for a page to render', async () => {
+  it('should return essential data for an on demand page to render', async () => {
```
Comment (RichardPK) [inline src/app/routes/onDemandAudio/getInitialData/index.test.js:19]:
```suggestion
  it('should return essential data for an on demand radio page to render', async () => {
```

File: src/app/routes/onDemandAudio/getInitialData/index.test.js [hunk 5]
```diff
@@ -40,7 +41,35 @@ describe('Get initial data for on demand radio', () => {
...
+      'ichef.bbci.co.uk/images/ic/$recipe/p02rt7vj.jpg',
+    );
+    expect(pageData.promoBrandTitle).toEqual('BBC Xtra');
+    expect(pageData.durationISO8601).toEqual('PT6M50S');
+    expect(pageData.thumbnailImageUrl).toEqual(
+      'https://ichef.bbci.co.uk/images/ic/1024x576/p02rt7vj.jpg',
+    );
+  });
+
   it('should return essential data for a page to render when the episode toggle is null', async () => {
```
Comment (RichardPK) [inline src/app/routes/onDemandAudio/getInitialData/index.test.js:71]:
```suggestion
  it('should return essential data for a page to render when the on demand radio episode toggle is null', async () => {
```

File: src/app/routes/onDemandAudio/getInitialData/index.test.js [hunk 6]
```diff
@@ -64,7 +93,8 @@ describe('Get initial data for on demand radio', () => {
     );
   });
 
-  it('should return recent episode data when recentEpisode toggle is enabled', async () => {
+  it('should return on demand recent episode data when recentEpisode toggle is enabled', async () => {
```
Comment (RichardPK) [inline src/app/routes/onDemandAudio/getInitialData/index.test.js:126]:
```suggestion
  it('should return on demand radio recent episode data when recentEpisode toggle is enabled', async () => {
```

**Review by RichardPK — APPROVED** [review]
Rad. Just some veeeeery nitpicky things from me yeoo

**Pull request merged**